### PR TITLE
Add TLS option to AMQP transport

### PR DESCRIPTION
**Pull request by bernd**
Adds a TLS option to AMQP transports. The transport connects to broker via SSL/TLS if enabled. This is a simple start without any certificate verification and not client certificate support. Tested with RabbitMQ.

Implementation for #850.

Objections?

### CODE REVIEW

File: graylog2-inputs/src/main/java/org/graylog2/inputs/transports/AmqpConsumer.java [hunk 1]
```diff
@@ -132,6 +136,15 @@ public void connect() throws IOException {
         factory.setPort(port);
         factory.setVirtualHost(virtualHost);
 
+        if (tls) {
+            try {
+                LOG.info("Enable TLS for AMQP input [{}/{}].", sourceInput.getName(), sourceInput.getId());
```
Comment (joschi) [inline graylog2-inputs/src/main/java/org/graylog2/inputs/transports/AmqpConsumer.java]: I'd either write "Enabled TLS […]" or "Enabling TLS […]".

File: graylog2-inputs/src/main/java/org/graylog2/inputs/transports/AmqpConsumer.java [hunk 2]
```diff
@@ -132,6 +136,15 @@ public void connect() throws IOException {
...
+                factory.useSslProtocol();
+            } catch (NoSuchAlgorithmException | KeyManagementException e) {
+                throw new IOException(e);
```
Comment (joschi) [inline graylog2-inputs/src/main/java/org/graylog2/inputs/transports/AmqpConsumer.java]: Maybe add a simple message like "Couldn't enable TLS for AMQP input." to the exception.

File: graylog2-inputs/src/main/java/org/graylog2/inputs/transports/AmqpTransport.java [hunk 3]
```diff
@@ -294,6 +297,15 @@ public ConfigurationRequest getRequestedConfiguration() {
                     )
             );
 
+            cr.addField(
+                    new BooleanField(
+                            CK_TLS,
+                            "Enable TLS?",
+                            false,
+                            "Enable transport encryption via TLS. (NEEDS correct TLS port setting!)"
```
Comment (joschi) [inline graylog2-inputs/src/main/java/org/graylog2/inputs/transports/AmqpTransport.java]: The all-caps word in the parentheses looks strange. Maybe rephrase to "Enable transport encryption via TLS (requires valid TLS port setting."

**Comment by bernd** [comment]
Thank you!
